### PR TITLE
Add the Efnysien survival rifle

### PR DESCRIPTION
**Pull request by blackrep**
[Game Objects][Feature]

This PR adds a new unique kinetic rifle to the game, the Efnysien survival rifle, produced by Mabinogi. This is a semi-automatic rifle is designed for Salvagers to be an alternative to the Salvager zip gun and has the unique ability to be converted to different calibers using barrel kits. The advantage this gun has over the salvager zip gun is that it uses the original capacity of the magazines provided to salvager for that caliber, and does not roll change to explode. Downside is that it is a 2 handed weapon, and to switch the gun to more powerful ammo, the user must purchase new barrel kits.

The PR includes:
1. The base rifle in .22 with proper inhand, item, and back_worn sprites
2. 3 barrel items which allow the user to swap between the different supported calibers

Sprites:
image

In-Game:
https://www.youtube.com/watch?v=X3yYUbu2k3M

## Why's this needed? 
Common feedback is that Salvagers need slightly more stronger offensive capability. This PR adds a new weapon to their arsenal which I believe will fit the role without stepping too far into the bounds of a Nukies reskin. 
image

IMO Salvagers should get access to the .22 barrel first -> 9mm -> 5.56 barrel last. Given the states of the rounds they're able to fire:
image

## Changelog 

```changelog
(u)Blackrep
(+)Salvagers have recently been spotted with some heavier firepower.
```

### CODE REVIEW

File: code/obj/item/gun/kinetic.dm [hunk 1]
```diff
@@ -483,6 +528,77 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic/single_action)
...
+/obj/item/gun/kinetic/survival_rifle
+	name = "\improper Efnysien survival rifle"
+	desc = "A semi-automatic rifle, renowned for it's easily convertible caliber, developed by Mabinogi Firearms Company."
+	icon = 'icons/obj/large/48x32.dmi'
+	icon_state = "survival_rifle_22"
+	item_state = "survival_rifle"
+	wear_state = "survival_rifle"
+	wear_image_icon = 'icons/mob/clothing/back.dmi'
+	force = MELEE_DMG_RIFLE
+	c_flags = NOT_EQUIPPED_WHEN_WORN | EQUIPPED_WHILE_HELD | ONBACK
```
Comment (Azrun) [inline code/obj/item/gun/kinetic.dm:540]: What are the first two flags for?

Comment (blackrep) [reply code/obj/item/gun/kinetic.dm:540]: Believe they're needed to properly allow worn on back behavior. They're used on every single gun that you can wear on your back:
https://github.com/goonstation/goonstation/blob/484bd57f3e09e2dbc22c126ea070f718fbf0107f/_std/defines/clothing.dm#L47-L50

Comment (RubberRats) [reply code/obj/item/gun/kinetic.dm:540]: Pretty sure those flags are for guns that have slowdown when wielded, like the nukie LMG. Assuming this one doesnt have slowdown, you should be OK to remove them.

Comment (blackrep) [reply code/obj/item/gun/kinetic.dm:540]: Interesting, I'll remove them and test.

Comment (blackrep) [reply code/obj/item/gun/kinetic.dm]: > Pretty sure those flags are for guns that have slowdown when wielded, like the nukie LMG. Assuming this one doesnt have slowdown, you should be OK to remove them.

Just tested, seems to have no discernable difference, so I'll just remove them.

**Pull request merged**